### DISCUSSION TITLE
Lazy variable scope

### DISCUSSION THREAD

**Discussion by Makistos** [AnWeber/httpyac#647] [General]
Not sure if my title is good, but bear with me.

I have set up some test files with some files calling tests in other files and I would like to be able to run all tests at once but also a single file.

An example:

I have a file called publishers.http which obviously tests publisher related stuff. One is publisher_create. The whole set does stuff like create a publisher, update it, test some errors and finally deletes the publisher.

I also have a pubseries.http which tests some series related stuff. This requires the use of publisher_create because series requires publisher id. File itself is similar but it also deletes the publisher as well because otherwise I would need to manually delete it if I run just this file.

But herein lies the problem. publisher_create is only run once even when I run all the files but publisher_delete is run twice. So one of the deletes causes an error.

I tried reading the docs but couldn't figure out how I could create a separate publisher in pubseries file, e.g. have it only scope that file.

So far the only solution I could come up with is a shell script that runs each file separately but that feels clunky.

P.S. If I could ask for a small feature: could the results in default mode (not JSON) also print the test name? I have close to 100 tests already and each file has 10-20 tests so when one fails it can be a bit hard to figure out which it was.

**Comment by AnWeber** [comment]
Can you just.scribble the main parts of the two files. I do not understand what mean with reqire publisher_create. Do you use @ref/ forceref?

**Comment by Makistos** [comment]
Ok, pubseries.http:
```
# @import ./login.http
# @import ./publishers.http

###
# @name pubseries_create
# @ref login
# @ref publisher_create
POST {{host}}/api/pubseries
Authorization: Bearer {{login.access_token}}

{
        "data": {
                "publisher_id": {{publisher_create}},
                "name": "Best of",
                "important": 1,
                "image_attr": "Wikipedia",
                "image_src": "http://wikipedia"
            }
    }
?? status == 201

###
# @ref login
# @name pubseries_delete
DELETE {{host}}/api/pubseries/{{pubseries_create}}
Authorization: Bearer {{login.access_token}}
?? status == 200

###
# @ref login
DELETE {{host}}/api/publishers/{{publisher_create}}
Authorization: Bearer {{login.access_token}}
?? status == 200
```

publisher.http
```
# @import ./login.http

###
# @ref login
# @name publisher_create
POST {{host}}/api/publishers
Authorization: Bearer {{login.access_token}}

{
    "data": {
            "name": "Foo Bar Kustannus",
            "fullname": "Foo Bar Kustannus Oy",
            "links": [
                {"link": "http://link.me",
                "description": "Link for Foo Bar Kustannus"}
            ]
        }
}
?? status == 201


```

**Comment by AnWeber** [comment]
Try changing it to `@forceRef pubseries_create`
